### PR TITLE
Feat-Watchlater-History-Likes

### PR DESCRIPTION
**Pull request by Shrinkhlarajpoot**
Features Added - 
Add to Watchlater
Delete from Watchlater
Like -Unlike
History-Delete from History

### CODE REVIEW

File: src/frontend/customHooks/useLikes.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        if (status === 201) {
+          toast.success("Added to Likes");
+          dispatchLikes({
+            type: "SET_LIKES",
+            payload: { Likeslist: data.likes },
+          });
+        }
+      } catch (error) {
+        toast.error("Try Again Later");
+        console.log(error);
```
Comment (manojasarsa) [inline src/frontend/customHooks/useLikes.jsx]: You can use `console.error` here as well.

**Review by manojasarsa — APPROVED** [review]
Rest. looks good.
GTG!

File: src/frontend/customHooks/useHistory.jsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        if (status === 200) {
+            toast.info("History Deleted");
+          dispatchWatchHistory({
+            type: "SET_WATCHHISTORY",
+            payload: { watchHistory: data.history },
+          });
+        }
+      } catch (error) {
+        toast.info("Try Again Later");
+        console.log(error);
```
Comment (msdeshmukh009) [inline src/frontend/customHooks/useHistory.jsx]: You can also show custom error messages depending on the status code. You can remove this console as you are handling the error with toasts.

File: src/frontend/pages/Explore/components/VideoCard.jsx [hunk 3]
```diff
@@ -1,14 +1,22 @@
...
 import { useAuth } from "../../../context";
+import { useHistory, useLikes, useWatchlater } from "../../../customHooks";
+
 const VideoCard = ({ video }) => {
   const [showPlaylistModal,setShowPlaylistModal] = useState(false);
-  const navigate = useNavigate();
-  const [showMoreOptionsModal, setShowMoreOptionsModal] = useState(false);
+  const currentroutePath = useLocation().pathname;
+ const navigate = useNavigate();
+ const [showMoreOptionsModal, setShowMoreOptionsModal] = useState(false);
```
Comment (msdeshmukh009) [inline src/frontend/pages/Explore/components/VideoCard.jsx]: We should try to put state declarations at the top followed by react inbuilt hooks and then custom hooks. By following this pattern the code will become more predictable.

File: src/frontend/reducers/likeReducer.jsx [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const likeReducerFun=(state,{payload,type})=>{
+    switch(type){
+        case "SET_LIKES":
+            return {...state, Likeslist:payload. Likeslist}
+        default:
+            return {...state}
```
Comment (msdeshmukh009) [inline src/frontend/reducers/likeReducer.jsx]: You can just return the state without destructing. I think that would also work.

File: src/frontend/routes/Pageroutes.jsx [hunk 5]
```diff
@@ -9,6 +9,7 @@ import {
   Signup,
   SinglePlaylist,
   SingleVideo,
+  WatchHistory,
   WatchLater,
 } from "../pages";
 // import { PrivateRoute } from "routes/Privateroutes";
```
Comment (msdeshmukh009) [inline src/frontend/routes/Pageroutes.jsx:15]: You can remove this comment if not required.

File: src/frontend/services/index.js [hunk 6]
```diff
@@ -9,3 +9,8 @@ export {
 } from "./playlistService"
 
 export {getNotesService,addNoteService,deleteNoteService} from "./notesService"
+export { getWatchLaterService,
+  addToWatchLaterService,
+  removeFromWatchLaterService,} from "./watchlaterService"
+export {getLikesService, addLikedVideoService, removeLikedVideoService} from "./likeService"
+export {getHistoryService,addToHistoryService,removeFromHistoryService,clearWatchHistoryService} from "./watchHistoryService"
```
Comment (msdeshmukh009) [inline src/frontend/services/index.js:16]: We can use prettier over here to format the code a little bit.

File: src/index.js [hunk 7]
```diff
@@ -22,9 +27,15 @@ ReactDOM.render(
...
             <PlaylistProvider>
+              <WatchlaterProvider>
+                <LikelistProvider>
+                  <WatchHistoryProvider>
               <NotesProvider>
               <App />
               </NotesProvider>
+              </WatchHistoryProvider>
+              </LikelistProvider>
+              </WatchlaterProvider>
```
Comment (msdeshmukh009) [inline src/index.js:38]: prettier over here as well

**Review by msdeshmukh009 — APPROVED** [review]
The code looks fine and the implementation is also working as expected. Good Work !!

Suggestion: You can navigate or show some message to the user when the user is not authenticated and is trying to access private features such as likes, and add to watch later.

File: src/frontend/customHooks/useHistory.jsx (hunk 2 above)
Comment (Shrinkhlarajpoot) [reply src/frontend/customHooks/useHistory.jsx]: Okay will modify it.Thanks

**Pull request merged**